### DISCUSSION TITLE
libnode.so

### DISCUSSION THREAD

**Discussion by mario6714** [1715173329/UnblockNeteaseMusic#1132] [General]
### 需求描述

Create libnode.so

### 预期行为

Someone know how to create a libnode.so from the most recent node js? thanks

### 网易云音乐版本号

7.3.28

### 操作系统

Android 13

### 其他信息

Someone know how to create a libnode.so from the most recent node js? thanks

**Comment by luoxingran** [comment]
This is a nodejs binary file, just change the file type to so support library, the purpose is to bypass Android 13 does not allow direct execution of binary files

**Reply by mario6714** [reply]
Hi, how i can do that?  Just rename (what file?) to .so?

**Reply by luoxingran** [reply]
https://github.com/luoxingran/Binary-nodejs-for-android
https://github.com/nining377/android-node-ci/releases/tag/v1.1.0

These are some nodejs binary files that have been cross-compiled. If you need other versions, you can also modify and compile them through nining377's project action.

**Reply by mario6714** [reply]
Thanks, i ask about how to create a recent libnode.so because i see in this project (i think) a recent node js and i eanted update that file for dolby beta and netease cloud music

**Reply by luoxingran** [reply]
You can try to change the parameters of build.sh in the nining377 project to trigger action for automatic compilation

**Reply by mario6714** [reply]
One question, for build i can use Android (termux) or i need Linux (example ubuntu)?

**Reply by mario6714** [reply]
I'm doing it wrong?

![Screenshot_20230708_221451_Termux](https://github.com/UnblockNeteaseMusic/server/assets/32179719/8f5dbb1a-0d1b-4cc5-a294-06247a7a2bf3)

**Reply by luoxingran** [reply]
You can fork nining377's project to directly change the data, using github ci to build or use linux for cross compilation

**Reply by mario6714** [reply]
Github ci? That new for me, is from windows or android or another way?

**Reply by mario6714** [reply]
Like this?

i use the option "Fork"

![imagen](https://github.com/UnblockNeteaseMusic/server/assets/32179719/6bbd4e54-890b-407d-b702-b9cb87fc09c2)

**Reply by mario6714** [reply]
Thank you for all of your help, if i have problem can i ask for your help one more time?

**Reply by mario6714** [reply]
I have this error:

![imagen](https://github.com/UnblockNeteaseMusic/server/assets/32179719/d8a333e0-c900-425d-ae56-aba581658181)


Maybe this help?

https://github.com/nodejs/node/issues/42888

**Reply by mario6714** [reply]
I cant find what i am missing

**Reply by luoxingran** [reply]
ci is not very perfect, there may be bugs when compiling different versions, and nodejs does not need to keep the latest version, if you want to use the latest version, you may use the https://github.com/dna2github/dna2oslab project to enter linux self-construction

**Reply by mario6714** [reply]
Ok, for example a libnode.so version 16.17.0 is ok right?

I already tried using a virtual machine using "ubuntu" but not luck

Thanks

**Reply by mario6714** [reply]
I really like dolby beta and Netease Cloud Music, but sometimes Dolby_beta stops working and I can't find why or how to fix the problem and I lose Netase Cloud Music and "Dolby_beta" module, already tried different apk (mods, different versions using Dolby_beta) but always without luck, do you happen to know of any other way (module or mods)?

And thanks for all the help, you help me a lot @luoxingran

**Reply by mario6714** [reply]
Can i make a libnode.so using a Ubuntu in VirtualBox machine?
